cloud_functions/climateiq_data_export_cf/main.py

```diff
@@ -302,20 +352,154 @@ def _calculate_h3_indexes(
         )
     ]
 
-    # TODO: Add logic for fetching predictions from neighboring chunks for boundary
-    # cells.
+    # Extract rows where the projected H3 cell is not fully contained within the chunk
+    # so we can aggregate prediction values across chunk boundaries.
+    boundary_h3_cells = spatialized_predictions[
+        spatialized_predictions.apply(
+            lambda row: not row["h3_boundary"].within(chunk_boundary),
+            axis=1,
+        )
+    ]["h3_boundary"].unique()
+
+    return _aggregate_h3_predictions(
+        study_area_metadata,
+        chunk_metadata,
+        boundary_h3_cells,
+        spatialized_predictions,
+        bucket_name,
+        object_name,
+        chunks_ref,
+    )
+
+
+def _aggregate_h3_predictions(
+    study_area_metadata: dict,
+    chunk_metadata: dict,
+    boundary_h3_cells: np.ndarray,
+    spatialized_predictions: pd.DataFrame,
+    bucket_name: str,
+    object_name: str,
+    chunks_ref: Any,
+) -> pd.Series:
+    """Aggregates predictions for duplicate H3 projections across chunk boundaries.
+
+    Args:
+        study_area_metadata: A dictionary containing metadata for the study
+        area.
+        chunk_metadata: A dictionary containing metadata for the chunk.
+        boundary_h3_cells: An array containing the H3 boundaries of cells that
+        intersect with the chunk boundary.
+        spatialized_predictions: A DataFrame containing H3 indexes and predictions
+        for a single chunk. This input DataFrame may contain duplicate H3 indexes.
+        bucket_name: The name of the GCS bucket containing model predictions.
+        object_name: The name of the chunk object this cloud function is currently
+        processing. Used to construct the GCS object name of the neighbor chunk.
+        chunks_ref: A reference to the chunks collection in Firestore, used to
+        retrieve metadata of the neighbor chunk.
+
+    Returns:
+        A Series containing H3 indexes in a single chunk along with associated
+        predictions, representing a subset of the full study area results.
+    """
+    x = chunk_metadata["x_index"]
+    y = chunk_metadata["y_index"]
+
+    # Process direct 8 neighbors of current chunk. Since H3 cells at resolution 13
+    # (43.870 m^2)have a smaller area than chunks (), it is guaranteed that any
+    # overlapping H3 cells from the current chunk are contained within these neighbors.
+    neighbors = {
+        (x - 1, y + 1),
+        (x, y + 1),
+        (x + 1, y + 1),
+        (x - 1, y),
+        (x + 1, y),
+        (x - 1, y - 1),
+        (x, y - 1),
+        (x + 1, y - 1),
+    }
+    for neighbor_x, neighbor_y in neighbors:
+        if (
+            neighbor_x < 0
+            or neighbor_y < 0
+            or neighbor_x >= study_area_metadata["col_count"]
+            or neighbor_y >= study_area_metadata["row_count"]
+        ):
+            # Chunk is outside the study area boundary.
+            continue
+        query = (
```

Comment:
This seems similar to _get_chunk_metadata. Is there a way to merge the two? Also, I'm confused why some chunk metadata is available in study_area_metadata and why some needs to be queried for in chunks_ref.

Reply:
study_area_metadata contains all the chunks in the study area but iterating through each chunk to get the neighbors would be inefficient. instead, we query firebase for specific x_index and y_index values

updated the validation of fields to happen in a shared helper method